qiskit/quantum_info/primitives/estimator/base_estimator.py

```diff
@@ -0,0 +1,158 @@
+
+
+class BaseEstimator(BasePrimitive, ABC):
+    """
+    Expectation Value class
+    """
+
+    def __init__(
+        self,
+        circuits: Union[QuantumCircuit, list[Union[QuantumCircuit]]],
+        observables: Union[BaseOperator, PauliSumOp, list[Union[BaseOperator, PauliSumOp]]],
+        backend: Union[Backend, BackendWrapper],
+        transpile_options: Optional[dict] = None,
+        bound_pass_manager: Optional[PassManager] = None,
+    ):
+        """ """
+        super().__init__(
+            backend=backend,
+            transpile_options=transpile_options,
+            bound_pass_manager=bound_pass_manager,
+        )
+
+        # Initialize quantum circuits
+        if isinstance(circuits, list):
+            self._circuits = [init_circuit(circuit) for circuit in circuits]
+        else:
+            self._circuits = [init_circuit(circuits)]
+
+        # Initialize observables
+        if isinstance(observables, list):
+            self._observables = [init_observable(observable) for observable in observables]
+        else:
+            self._observables = [init_observable(observables)]
+
+        self._grouping = [
+            Group(i, j) for i in range(len(self._circuits)) for j in range(len(self._observables))
+        ]
+        self._transpiled_circuits_cache: dict[tuple[Group], list[QuantumCircuit]] = {}
+
+    @property
+    def circuits(self) -> list[QuantumCircuit]:
+        """Quantum Circuits that represents quantum states.
+
+        Returns:
+            quantum states
+        """
+        return self._circuits
+
+    @property
+    def observables(self) -> list[SparsePauliOp]:
+        """
+        SparsePauliOp that represents observable
+
+        Returns:
+            observable
+        """
+        return self._observables
+
+    @property
+    def grouping(self) -> list[Group]:
+        """
+        Group of circuits and observables.
+
+        Returns:
+            grouping
+        """
+        return self._grouping
+
+    def set_transpile_options(self, **fields) -> BaseEstimator:
+        """Set the transpiler options for transpiler.
+
+        Args:
+            fields: The fields to update the options
+        Returns:
+            self
+        """
+        self._transpiled_circuits = None
+        self._transpiled_circuits_cache = {}
+        super().set_transpile_options(**fields)
+        return self
+
+    @property
+    def preprocessed_circuits(
+        self,
+    ) -> PreprocessedCircuits:
+        """
+        Transpiled quantum circuits produced by preprocessing
+
+        Returns:
+            List of the transpiled quantum circuit
+        """
+        self._preprocessed_circuits = self._preprocessing(self.circuits, self.observables)
+        return super().preprocessed_circuits
+
+    # pylint: disable=arguments-differ
+    def run(
+        self,
+        parameters: Optional[Union[list[float], list[list[float]]]] = None,
```

Comment:
Currently estimator accepts parameters as list of floats.

I have a comment on parameters resolving. 
For example I have 2 circuits with parameters (first circuit has `theta[0]`, `theta[1]` parameters and second one has `SOME_RANDOM_NAME` parameter):
```python
psi1 = TwoLocal(...)
psi2 = QuantumCircuit(...)
psi2.u1(Parameter("SOME_RANDOM_NAME"))
...
with Estimator(circuits=[psi1, psi2], observables=[H1]) as e:
    res = e([0, 1, 2])
```
How estimator will decide which parameter to bind to which circuit? And this example is without `grouping` (easy one). 

Maybe we should ~~also~~ only accept mappings to be specific in parameters? Like:
```python
psi1 = TwoLocal(...)
psi2 = QuantumCircuit(...)
psi2.u1(Parameter("SOME_RANDOM_NAME"))
...
with Estimator(circuits=[psi1, psi2], observables=[H1]) as e:
    res = e({"theta[0]": 0, "theta[1]": 1, "SOME_RANDOM_NAME": 2})
```
In that case we also need to fix `bind_parameters` method in `QuantumCIrcuit` to skip parameters that are not existing in circuit but were passed in mapping. Right now if you pass something extra circuit will rase `CircuitError: If values is a dict and contains parameters not present in the circuit.`